tests/presentation/windows/tst_MainWindowViewModel.cpp

```diff
@@ -0,0 +1,319 @@
+                expectedRequest->getHeaders())
+        ).RETURN(expectedRequest);
+
+        REQUIRE_CALL(*requestService, send(expectedRequest))
+            .RETURN(std::async(std::launch::async, [] { return std::make_shared<domain::models::Response>(); }));
+
+        // Act
+        sut->sendRequest(
+                expectedRequest->getMethod(),
+                expectedRequest->getUri(),
+                expectedRequest->getHeaders(),
+                expectedRequest->getBody());
+    }
+
+    SECTION("updates the state to Sent when the request is sent")
+    {
+        // Arrange
+        auto view = std::make_shared<MainWindowStub>();
+        auto requestService = std::make_shared<RequestServiceMock>();
+        auto requestRepository = std::make_shared<RequestRepositoryMock>();
+
+        const auto& sut = std::make_unique<MainWindowViewModel>(requestFactory, requestServiceFactory,
+                                                                requestRepositoryFactory, view);
+
+        auto expectedResponse = std::make_shared<domain::models::Response>();
+        expectedResponse->body = "Success!";
+
+        ALLOW_CALL(*requestServiceFactory, getRequestService()).RETURN(requestService);
+        ALLOW_CALL(*requestRepositoryFactory, getRepository()).RETURN(requestRepository);
+        ALLOW_CALL(*requestFactory, getRequest(trompeloeil::_, trompeloeil::_, trompeloeil::_))
+            .RETURN(std::make_shared<domain::models::Request>());
+        ALLOW_CALL(*requestService, send(trompeloeil::_))
+            .RETURN(std::async(std::launch::async, [expectedResponse] { return expectedResponse; }));
+
+        // Act
+        sut->sendRequest("", "", {}, nullptr);
+
+        // Assert
+        REQUIRE(std::dynamic_pointer_cast<presentation::states::Sent>(view->state));
+
+        const auto& actualState = std::dynamic_pointer_cast<presentation::states::Sent>(view->state);
+        REQUIRE(actualState->response->body == expectedResponse->body);
+    }
+
+    SECTION("updates the state to Error when an exception is thrown")
+    {
+        // Arrange
+        auto view = std::make_shared<MainWindowStub>();
+        auto requestService = std::make_shared<RequestServiceMock>();
+        auto requestRepository = std::make_shared<RequestRepositoryMock>();
+
+        const auto& sut = std::make_unique<MainWindowViewModel>(requestFactory, requestServiceFactory,
+                                                                requestRepositoryFactory, view);
+
+        ALLOW_CALL(*requestServiceFactory, getRequestService()).RETURN(requestService);
+        ALLOW_CALL(*requestRepositoryFactory, getRepository()).RETURN(requestRepository);
+        ALLOW_CALL(*requestFactory, getRequest(trompeloeil::_, trompeloeil::_, trompeloeil::_))
+            .RETURN(std::make_shared<domain::models::Request>());
+        ALLOW_CALL(*requestService, send(trompeloeil::_))
+            .THROW(data::exceptions::NoAvailableRepositoryException());
```

Comment:
Use an actual exception that could possibly be thrown in the RequestRepository to make the test case more comprehensive